core/app/models/spree/promotion.rb

```diff
@@ -67,6 +67,19 @@ def self.with_coupon_code(val)
       ).first
     end
 
+    # All orders that have been discounted using this promotion
+    def discounted_orders
```

Comment:
This looks like a relationship, I'm wondering if it would be preferable to define it that way... it looks possible to me, but I think we'd need to add a few relationships on other models in that case. What do you think?

Reply:
I'm a bit wary of defining a large amount of relationships. They offer a bunch of new APIs that I would not want to exist: 
`order.applied_promotions.create!(...)`? Doesn't look like a step forward to me. This simple method does the job :)